packages/calcite-components/src/components/swatch/swatch.tsx

```diff
@@ -0,0 +1,465 @@
+import { PropertyValues } from "lit";
+import { createRef } from "lit-html/directives/ref.js";
+import {
+  LitElement,
+  property,
+  createEvent,
+  h,
+  method,
+  state,
+  JsxNode,
+  Fragment,
+} from "@arcgis/lumina";
+import Color, { ColorInstance } from "color";
+import { slotChangeHasAssignedElement } from "../../utils/dom";
+import { Scale, SelectionMode } from "../interfaces";
+import {
+  InteractiveComponent,
+  InteractiveContainer,
+  updateHostInteraction,
+} from "../../utils/interactive";
+import { useSetFocus } from "../../controllers/useSetFocus";
+import type { SwatchGroup } from "../swatch-group/swatch-group";
+import { hexify } from "../color-picker/utils";
+import { CHECKER_DIMENSIONS } from "../color-picker-swatch/resources";
+import { CSS, SLOTS, IDS } from "./resources";
+import { styles } from "./swatch.scss";
+
+declare global {
+  interface DeclareElements {
+    "calcite-swatch": Swatch;
+  }
+}
+
+export class Swatch extends LitElement implements InteractiveComponent {
+  //#region Static Members
+
+  static override styles = styles;
+
+  //#endregion
+
+  //#region Private Properties
+
+  private internalColor: ColorInstance;
+
+  private containerEl = createRef<HTMLDivElement>();
+
+  private focusSetter = useSetFocus<this>()(this);
+
+  //#endregion
+
+  //#region State Properties
+
+  @state() private hasImage = false;
+
+  //#endregion
+
+  //#region Public Properties
+
+  /**
+   * Specifies the component's color
+   *
+   * @see https://developer.mozilla.org/en-US/docs/Web/CSS/color_value
+   */
+  @property() color: string | null;
```

Comment:
Should this maybe be `  @property() color?: string;`? We don't need to reflect this right?